coremltools/models/utils.py

```diff
@@ -1153,15 +1153,24 @@ def updateBlobFileName(proto_message, new_path):
         map(lambda spec: spec.specificationVersion, input_specs)
     )
 
-    # Set pipeline input
-    pipeline_spec.description.input.MergeFrom(
-        input_specs[0].description.input
-    )
-
-    # Set pipeline output
-    pipeline_spec.description.output.MergeFrom(
-        input_specs[-1].description.output
-    )
+    # If a later model doesn't get an input from a previous model, it must be
+    # an input to the pipeline.
+    available_as_input = set()
+    for cur_spec in input_specs:
+        for cur_input in cur_spec.description.input:
+            if cur_input.name not in available_as_input:
+                pipeline_spec.description.input.add().MergeFrom(cur_input)
+                available_as_input.add(cur_input.name)
```

Comment:
`cur_input` is added to `available_as_input` because we don't want to add it again to the inputs for the pipeline. We don't want duplicate entries in the inputs for the pipeline.

I think input/output names, for models in a pipeline, are assumed to be unique in a pipeline. In your example the `x` from `model_1` and `model_2` would be the same input, not different inputs with the same name. I'm testing something similar in the `test_pipeline_outputs_from_multiple_models` unit test of this PR (`y1` variable).

Reply:
`test_pipeline_outputs_from_multiple_models ` 's `y1` comes from the output of the first model.
Might worth to add a test in which `y1` is the actually pipeline input so that this test can cover this line:
`available_as_input.add(cur_input.name)`